Sources/Components/CarouselView/CarouselView.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public typealias CarouselViewCell = UICollectionViewCell
+
+public protocol CarouselViewDataSource: AnyObject {
+    func numberOfItems(in carouselView: CarouselView) -> Int
+    func carouselView(_ carouselView: CarouselView, cellForItemAt indexPath: IndexPath) -> CarouselViewCell
+}
+
+public protocol CarouselViewDelegate: AnyObject {
+    func carouselView(_ carouselView: CarouselView, didSelectItemAt indexPath: IndexPath)
+    func carouselViewDidEndDecelerating(_ carouselView: CarouselView)
```

Comment:
Should this function be required in protocol implementations?

Reply:
We do use this method in the `AdPageGalleryView`, that is why I added it. Should I make an empty extension?

Reply:
No, I think it's better to keep it then.